referenceframe/frame_system.go

```diff
@@ -397,6 +401,44 @@ func (sfs *simpleFrameSystem) getFrameToWorldTransform(inputMap map[string][]Inp
 	return srcToWorld, err
 }
 
+// ReplaceFrame transfers replaceMe's children and parentage to replaceWith. ReplaceMe is removed entirely from the frame system.
+// ReplaceWith is not allowed to exist within the frame system at the time of the call.
+func (sfs *simpleFrameSystem) ReplaceFrame(fs FrameSystem, replaceMe, replaceWith Frame) error {
+	if len(fs.FrameNames()) == 0 {
+		return errors.New("cannot replace frames within an empty framesystem")
+	}
+	if replaceMe.Name() == replaceWith.Name() {
+		return errors.New("replaceWith cannot have the same name as replaceMe")
+	}
+	if fs.Frame(replaceWith.Name()) != nil {
+		return errors.New("replaceWith is not allowed to exist within the framesystem at the time of this call")
+	}
+
+	// get replaceMe's parent
+	replaceMeParent, err := fs.Parent(replaceMe)
+	if err != nil {
+		return err
+	}
+
+	for f, parent := range sfs.parents {
+		// if frame is replaceMe remove it entirely
+		if f == replaceMe {
```

Comment:
I don't think you can do this until the end of the function?

Reply:
it should be safe as we've already gotten `replaceMe`'s parent.
Further, we shouldn't use `RemoveFrame`

Reply:
Wont other things potentially have this as its parent?  When you remove it it will have a nil parent.  Maybe I'm thinking about this wrong though

Reply:
@raybjork that'll be dealt with below, with `			sfs.parents[f] = replaceWith`

Reply:
I see we're just removing the replaceMe from the maps not freeing the memory in the "C" sense